### ISSUE TITLE
MoE models that balance with a router bias never expose their router logits

### ISSUE THREAD

**Issue by qgallouedec** [huggingface/transformers#48911]
### System Info

- `transformers` version: 5.17.0.dev0
- Platform: Linux-6.8.0-1057-aws-x86_64-with-glibc2.35
- Python version: 3.13.13
- Huggingface_hub version: 1.29.0
- Accelerate version: 1.14.0
- PyTorch version: 2.13.0+cu130 (GPU: True)

### Who can help?

@ArthurZucker @Cyrilvallez @vasqu

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [x] An officially supported task in the `examples` folder
- [ ] My own task or dataset

### Reproduction

17 MoE models compute router logits and then drop them. They never leave the forward, so there is no
way to get at expert routing from the outside:

```
axk1 axk2 deepseek_v3 deepseek_v32 dots1 exaone_moe glm4_moe glm4_moe_lite glm_moe_dsa hy_v4
kimi_linear lfm2_moe longcat_flash mimo_v2_flash nemotron_h solar_open step3p7
```

The logits exist. DeepSeek-V2's gate computes them on line 100 and returns them on line 117:

```python
router_logits = F.linear(hidden_states.type(torch.float32), self.weight.type(torch.float32))
...
return router_logits, topk_weights, topk_indices
```

They just never make it into the model output. `DeepseekV3ForCausalLM` returns a plain
`CausalLMOutputWithPast`, so `outputs.router_logits` does not exist and neither does `aux_loss`.

These are all models that balance their experts with a per-expert bias rather than an auxiliary loss,
so not computing the loss is correct. Withholding the logits is a separate thing.

### Expected behavior

Return `router_logits` from these models when `output_router_logits=True`, the way Qwen3-MoE and the
rest already do. No behaviour change by default, no new loss, just carrying a tensor up that the gate
already produces.

Two things it unblocks:

1. Inspecting expert utilization and load during training or eval, which today needs a forward hook.
2. Letting a trainer compute its own balancing term. TRL does this in its chunked SFT path, calling `load_balancing_loss_func` on `outputs.router_logits`. For these 17 the user can ask for the auxiliary loss and there is nothing to compute it from, so the request fails.

The second point is why this matters to us. A user fine-tuning DeepSeek-V3 who decides they want the loss has no path to it, even though the routing tensors are right there.

Related: #48885 wires the flag through the MoE VLM wrappers, #48886 covers models advertising an `aux_loss` they never set, #48889 is Llama 4's dead config fields.

**Comment by qgallouedec** [comment]
this issue is not opened to external contributions
